packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -860,7 +858,7 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
         );
 
         this.snapshotTree = props.snapshotTree;
-        this.isRootDataStore = props.isRootDataStore ?? false;
+        this.isLocalRootDataStore = props.isRootDataStore;
```

Comment:
I do not know if it will work, but can we
- call this.setRoot() if props.isRootDataStore === true here.
- remove this.isLocalRootDataStore altogether, and all the code that deals with it
   - createAttributes() below to use this.isAliased() (unfortunately it cannot call this.isRoot() due to async nature)
- getInitialSnapshotDetails() always return isRootDataStore: false
    - I believe it should not matter, because only user of it is isRoot() function that will already OR it with value of this.isAliasedDataStore set above
- remove setRoot() overwrite - parent implementation is enough

Reply:
This seems like a good plan. Local data stores only care about the if its alias or not. If its created as root, we still call it an alias which isn't exactly true but its okay. Remote data stores care about both initial snapshot state and aliasing.